microsoft/testsuites/security/fips.py

```diff
@@ -32,61 +137,84 @@ class Fips(TestSuite):
-                    )
-        else:
-            result = node.execute("command -v fips-mode-setup", shell=True)
-            if result.exit_code != 0:
-                raise SkippedException(
-                    "Command not found: fips-mode-setup. "
-                    f"Please ensure {node.os.name} supports fips mode."
-                )
-
-            node.execute("fips-mode-setup --enable", sudo=True)
-
-            log.info("FIPS mode set to enable. Attempting reboot.")
-            node.reboot()
-
-            result = node.execute("fips-mode-setup --check")
-
-            assert_that(result.stdout).described_as(
-                "FIPS was not properly enabled."
-            ).contains("is enabled")
+        result = node.execute("command -v fips-mode-setup", shell=True)
+        if result.exit_code != 0:
+            raise SkippedException(
+                "Command not found: fips-mode-setup. "
+                f"Please ensure {node.os.name} supports fips mode."
+            )
+
+        node.execute("fips-mode-setup --enable", sudo=True)
+
+        log.info("FIPS mode set to enable. Attempting reboot.")
+        node.reboot()
+
+        result = node.execute("fips-mode-setup --check")
+
+        assert_that(result.stdout).described_as(
+            "FIPS was not properly enabled."
+        ).contains("is enabled")
+
+    def _ensure_fips_expectations(
+        self, log: Logger, node: Node, variables: Dict[str, Any], should_be_fips: bool
+    ):
+        """
+        Ensures that the expectations about the node's FIPS status are correct
+        for the test.
+        The argument `should_be_fips` indicates whether the test should be run on a
+        FIPS vs. non-FIPS image.
+        To determine whether the image is FIPS or not, the function first checks
+        the `testing_fips_image` variable in the `variables` dictionary.
+        If it can't determine the image type from the variable, it falls back to checking
+        the image SKU from the azure metadata endpoint.
+        If this does not match the expectation, a SkippedException is raised.
+
+        Args:
+            log (Logger): The logger instance for logging messages.
+            node (Node): The node object representing the target machine.
+            should_be_fips (bool): A flag indicating whether the test should be
+                                run on a FIPS vs. non-FIPS image.
+            variables (Dict[str, Any]): A dictionary of variables containing the
+                                    'testing_fips_image' key.
+        Raises:
+            SkippedException: If the FIPS image expectation does not match the actual image SKU.
+        """
+        log.debug(f"ensure_fips_expectations: should_be_fips is '{should_be_fips}'")
+        log.debug(f"ensure_fips_expectations: variables is '{variables}'")
+
+        # First, try to deduce the FIPS image type from the variables dictionary.
+        fips_image_map = {"yes": True, "no": False}
```

Comment:
If use "true/false" as input, it should be more straightforward.

Reply:
No yet, can you create one in util/__init__.py like this? https://stackoverflow.com/questions/715417/converting-from-a-string-to-boolean-in-python